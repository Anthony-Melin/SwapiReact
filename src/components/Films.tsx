import { PropsWithChildren } from 'react'

type FilmsProps = PropsWithChildren<{ values: string[] }>

const Films = ({ values, ...restProps }: FilmsProps) => {
    return (
        <div datatest-id="films" {...restProps}>
            <p>Films</p>
            <ul>
                {values.map((value, index) => {
                    return <li datatest-id={`film-${index}`}>{value}</li>
                })}
            </ul>
        </div>
    )
}

export default Films
