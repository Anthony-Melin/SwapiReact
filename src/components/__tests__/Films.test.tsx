import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import Films from '../Films.tsx'

describe('Films', () => {
    describe('with empty list', () => {
        beforeEach(() => {
            render(<Films values={[]} />)
        })

        it('should match snapshot', () => {
            expect(screen.getByTestId('films')).toMatchSnapshot()
        })

        it('should have a title', () => {
            expect(screen.getByTestId('films')).toHaveTextContent('Films')
        })

        it('should have empty list label', () => {
            expect(screen.getByTestId('films-empty')).toHaveTextContent('No films linked')
        })
    })

    describe('with filled list', () => {
        beforeEach(() => {
            render(<Films values={['VALUE_1', 'VALUE_2']} />)
        })

        it('should match snapshot', () => {
            expect(screen.getByTestId('films')).toMatchSnapshot()
        })

        it('should have a title', () => {
            expect(screen.getByTestId('films')).toHaveTextContent('Films')
        })

        it('should contains elements', () => {
            expect(screen.getByTestId('films-0')).toHaveTextContent('VALUE_1')
            expect(screen.getByTestId('films-1')).toHaveTextContent('VALUE_2')
        })
    })
})
