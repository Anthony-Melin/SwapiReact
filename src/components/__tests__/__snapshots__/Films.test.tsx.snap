// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Films > with empty list > should match snapshot 1`] = `
<div
  data-testid="films"
>
  <p
    data-testid="films-title"
  >
    Films
  </p>
  <p
    data-testid="films-empty"
  >
    No 
    films
     linked
  </p>
</div>
`;

exports[`Films > with filled list > should match snapshot 1`] = `
<div
  data-testid="films"
>
  <p
    data-testid="films-title"
  >
    Films
  </p>
  <ul>
    <li
      data-testid="films-0"
    >
      VALUE_1
    </li>
    <li
      data-testid="films-1"
    >
      VALUE_2
    </li>
  </ul>
</div>
`;
